Sources/SwiftParserDiagnostics/ParseDiagnosticsGenerator.swift

```diff
@@ -1120,6 +1120,33 @@ public class ParseDiagnosticsGenerator: SyntaxAnyVisitor {
     return .visitChildren
   }
 
+  public override func visit(_ node: LabeledSpecializeEntrySyntax) -> SyntaxVisitorContinueKind {
+    if shouldSkip(node) {
+      return .skipChildren
+    }
+
+    if let unexpectedIdentifier = node.unexpectedBeforeLabel?.onlyPresentToken(where: { $0.tokenKind.isIdentifier }) {
+      addDiagnostic(
+        unexpectedIdentifier,
+        UnknownParameterError(
+          parameter: unexpectedIdentifier,
+          validParameters: [
+            .keyword(.target),
+            .keyword(.availability),
+            .keyword(.exported),
+            .keyword(.kind),
+            .keyword(.spi),
+            .keyword(.spiModule),
+            .keyword(.available),
```

Comment:
This is copied in. 

Should try to look into if we could use `enum SpecializeParameter: TokenSpecSet {}` for this? 

And second: Should we generate `SpecializeParameter` based on the label.choices in `AttributeNodes.swift`?

Reply:
> Should try to look into if we could use `enum SpecializeParameter: TokenSpecSet {}` for this?

Yes, there is some precedence for that already. Look for the `TokenSpecSet` that are marked as `@_spi(RawSyntax)`. I think you could very reasonably move `SpecializeParameter` into TokenSpecSet.swift.



> And second: Should we generate `SpecializeParameter` based on the label.choices in `AttributeNodes.swift`?

That has been a dream of mine for a while now, to generate `TokenSpecSet` based on the node choices. If you would like to look into that, that would be great.